enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1720,12 +1729,24 @@ func (p *PebbleCache) deleteFileAndMetadata(ctx context.Context, key filestore.P
 		return status.FailedPreconditionErrorf("Unnown storage metadata type: %+v", storageMetadata)
 	}
 
-	if md.GetStoredSizeBytes() > 0 {
-		p.sendSizeUpdate(partitionID, key.CacheType(), -1*md.GetStoredSizeBytes())
+	delta := md.GetStoredSizeBytes()
+	if p.includeMetadataSize {
+		delta = getTotalSizeBytes(md)
 	}
+	p.sendSizeUpdate(partitionID, key.CacheType(), -1*delta)
```

Comment:
What do you think about changing sendSizeUpdate to take `md` and 'op' (add or delete) as parameters so that this logic lives in a single place?

Reply:
Done.